conda_env/utils/binstar.py

```diff
@@ -0,0 +1,107 @@
+from binstar_client import errors
+from binstar_client.commands.login import interactive_login
+from collections import namedtuple
+from binstar_client.utils import get_binstar
+from conda_env.utils.spec import Spec
+from conda.cli import common
+from conda.resolve import normalized_version
+
+
+ENVIRONMENT_TYPE = 'env'
+
+# TODO: refactor binstar so that individual arguments are passed in instead of an arg object
+binstar_args = namedtuple('binstar_args', ['site', 'token'])
+
+def ensure_loggedin(binstar, token=None):
+    """
+    Ensure that a user is logged in to binstar. 
+    If token argument is not given, 
+    run an interactive prompt to log the user in if required.
+    """
+    try:
+        return binstar.user()
+    except errors.Unauthorized:
+        if not token:
+            print('The action you are performing requires authentication, '
+                  'please sign in to binstar:')
+            interactive_login(binstar_args(None, None))
+
+            # TODO: refactor binstar to make this easier e.g `binstar.reload_token() or binstar.ensure_login()`
+            # In binstar cli the binstar object is re-created.
+            binstar.token = get_binstar().token
+            binstar._session.headers.update({'Authorization': 'token %s' % (binstar.token)})
+
+            return binstar.user()
+        else:
+            raise
```

Comment:
+1 to a more clear error message.  Actually, if you raise an exception that extends from `RuntimeError`, that will be formatted and output [inside conda][1].  Let's do that.  There's a `conda_env.exceptions` module with tests around what should happen.  Add one to that that extends from `CondaEnvException` and `RuntimeError` and you're set.

[1]: https://github.com/conda/conda/blob/master/conda/cli/main.py#L212-L213